File: core/src/main/java/uk/co/amlcurran/social/EventCalendarItem.java
package uk.co.amlcurran.social;

class EventCalendarItem implements CalendarItem {
    private final String eventId;
    private final String title;
    private final Timestamp start;
    private final Timestamp endTime;

    public EventCalendarItem(String eventId, String title, Timestamp time, Timestamp endTime) {
        this.eventId = eventId;
        this.title = title;
        this.start = time;
        this.endTime = endTime;
    }

    public String id() {
        return eventId;
    }

    @Override
    public String title() {
        return title;
    }

    @Override
    public Timestamp startTime() {
        return start;
    }

    @Override
    public Timestamp endTime() {
        return endTime;
    }

    @Override
    public boolean isEmpty() {
        return false;
    }
}

File: android/shared/src/commonMain/kotlin/uk/co/amlcurran/social/EventCalendarItem.kt
package uk.co.amlcurran.social

import kotlinx.collections.immutable.ImmutableList
import kotlinx.datetime.Instant

data class EventCalendarItem(
    val eventId: String,
    val calendarId: String,
    override val title: String,
    override val startTime: Instant,
    override val endTime: Instant,
    val attendees: ImmutableList<Attendee>
) : CalendarItem {

    override var isEmpty: Boolean = false
}
